src/rbus/rbus_interval_sub.c

```diff
@@ -0,0 +1,280 @@
+
+        RBUSLOG_DEBUG("rbusEvent_Publish: publishing event %s to listener %s interval %d", sub->eventName, sub->listener, sub->interval);
+        error = rbus_publishSubscriberEvent(
+                handleInfo->componentName,
+                sub->eventName/*use the same eventName the consumer subscribed with; not event instance name eventData->name*/,
+                sub->listener,
+                msg);
+
+        rbusMessage_Release(msg);
+
+        rbusObject_Release(data);
+
+        if(error != RBUSCORE_SUCCESS)
+        {
+            RBUSLOG_ERROR("%s: rbusEvent_Publish failed with result=%d", __FUNCTION__, result);
+        }
+
+        /*update the record's property with new value*/
+        rbusProperty_SetValue(sub_rec->property, rbusProperty_GetValue(property));
+        rbusProperty_Release(property);
+        count++;
+    }
+    ERROR_CHECK(pthread_mutex_unlock(&sub_rec->mutex));
+    RBUSLOG_DEBUG("%s: stop\n", __FUNCTION__);
+    return NULL;
+}
+
+/* Add Subscription Record to global record and start subscription thread */
+void AddSubscriptionRecord(
+        rbusHandle_t handle,
+        elementNode* propNode,
+        rbusSubscription_t* sub)
+{
+    sRecord *sub_rec = NULL;
+    if(!propNode)
+    {
+        RBUSLOG_ERROR("%s: propNode NULL error", __FUNCTION__);
+        return;
+    }
+
+    if(propNode->type != RBUS_ELEMENT_TYPE_PROPERTY)
+    {
+        RBUSLOG_ERROR("%s: propNode type %d error", __FUNCTION__, propNode->type);
+        return;
+    }
+    assert(propNode->cbTable.getHandler);
+    if(!propNode->cbTable.getHandler)
+    {
+        RBUSLOG_ERROR("%s: propNode getHandler NULL error", __FUNCTION__);
+        return;
+    }
+
+    ERROR_CHECK(pthread_mutex_lock(&gMutex));
+    if (!gRecord)
+        rtVector_Create(&gRecord);
+
+    sub_rec = sub_find(sub);
+    ERROR_CHECK(pthread_mutex_unlock(&gMutex));
+    if(!sub_rec)
+    {
+        sub_rec = (sRecord*)rt_malloc(sizeof(sRecord));
+        sub_rec->handle = handle;
+        sub_rec->node = propNode;
+        sub_rec->sub = sub;
+        rbusProperty_Init(&sub_rec->property, propNode->fullName, NULL);
+        init_thread(sub_rec);
+
+        rtVector_PushBack(gRecord, sub_rec);
+
+        /* Thread create */
+        sub_rec->running = 1;
+        pthread_create(&sub_rec->thread, NULL, PublishingThreadFunc, (void *)sub_rec);
+    }
+}
+
+/* Delete Subscription Record from global record and stop it's running thread */
+void RemoveSubscriptionRecord(
+        rbusHandle_t handle,
+        elementNode* propNode,
+        rbusSubscription_t* sub)
+{
+    (void)(handle);
+    (void)(propNode);
+    VERIFY_NULL(sub);
+
+    ERROR_CHECK(pthread_mutex_lock(&gMutex));
+    if (!gRecord)
+    {
+        return;
+    }
+
+    sRecord* rec;
+    rec = sub_find(sub);
+    ERROR_CHECK(pthread_mutex_unlock(&gMutex));
+
+    if(rec)
+    {
+        if(rec->running)
```

Comment:
Use `rec->mutex` before you set `rec->running` to false;

Reply:
Working fine without adding 'rec->mutex' lock